### PR TITLE
feat: add documentation for setting up NetID provider

### PR DESCRIPTION
**Pull request by piotrmsc**
## Related Issue or Design Document

Added documentation describing how to add NetID as Social Identity Provider to Kratos.

This is a complement to https://github.com/ory/kratos/pull/2394 and should be merged after linked PR.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Review by tomekpapiernik — APPROVED** [review]

File: docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx [hunk 1]
```diff
@@ -1155,6 +1155,91 @@ selfservice:
...
+To set up "Sign in with NetID" you must create a
+[NetID Service](https://developerzone.netid.dev/devportal/tutorial/services/).
+
+Next you muste create a
+[NetID Client](https://developerzone.netid.dev/devportal/tutorial/clients/).
+
+Set the "Redirect URI" to:
+
+```
+https://playground.projects.oryapis.com/api/kratos/public/self-service/methods/oidc/callback/gitlab
```
Comment (aeneasr) [inline docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx:1169]: Gitlab? 😅

File: docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx [hunk 2]
```diff
@@ -1155,6 +1155,91 @@ selfservice:
...
+The pattern of this URL is:
+
+```
+http(s)://<domain-of-ory-kratos>:<public-port>/self-service/methods/oidc/callback/<provider-id>
+```
+
+:::note
+
+While you can use
+[NetID as an OIDC identity provider](https://docs.gitlab.com/ee/integration/openid_connect_provider.html),
```
Comment (aeneasr) [inline docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx:1181]: gitlab

File: docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx [hunk 3]
```diff
@@ -1155,6 +1155,91 @@ selfservice:
...
+# kratos -c path/to/my/kratos/config.yml serve
+selfservice:
+  methods:
+    oidc:
+      enabled: true
+      config:
+        providers:
+          - id: netid # this is `<provider-id>` in the Authorization callback URL. DO NOT CHANGE IT ONCE SET!
+            provider: netid
+            client_id: .... # Replace this with the OAuth2 Client ID provided by GitLab
```
Comment (aeneasr) [inline docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx:1222]: gitlab

File: docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx [hunk 4]
```diff
@@ -1155,6 +1155,91 @@ selfservice:
...
+            client_secret: .... # Replace this with the OAuth2 Client Secret provided by GitLab
```
Comment (aeneasr) [inline docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx:1223]: gitlab

File: docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx [hunk 5]
```diff
@@ -1155,6 +1155,91 @@ selfservice:
...
+            mapper_url: file:///etc/config/kratos/oidc.gitlab.jsonnet
```
Comment (aeneasr) [inline docs/kratos/guides/sign-in-with-github-google-facebook-linkedin.mdx:1224]: gitlab

**Review by aeneasr — CHANGES_REQUESTED** [review]

**Comment by piotrmsc** [comment]
One does not simply type Gitlab 😅  thx @aeneasr, please take a look again :)

**Review by aeneasr — APPROVED** [review]

**Pull request merged**